HW02_A_Graph-Fails.Rmd

```diff
@@ -1,87 +1,104 @@
+
+It is also best coding practice to only call the packages you use, so if you use a package but end up tossing the code you use for it, then make sure to remove loading it in the first place. For example, I could use `library("tidyverse")` but since this script will only be using ggplot2, I only load ggplot2.
+
+```{r load libraries}
+library("ggplot2")
+library("magrittr") #so I can do some piping
+```
+
+
+### Graph Fail 1
+
+What error is being thrown? How do you correct it? (hint, the error message tells you)
+
+ANSWER: The error being thrown says "Error: 'mapping' mus be created by 'aes()'" as well as "Did you use %>% instead of +?". Additionally, the variable name does not appear to be "city" but "cty", although this was not thrown as an error until the others were fixed, which I did below:
+
+```{r}
+data(mpg) #this is a dataset from the ggplot2 package
+
+mpg %>% 
+  ggplot(mapping = aes(x = cty, y = hwy, color = "blue")) + 
+  geom_point()
+
+```
+
+### Graph Fail 2
+Why aren't the points blue? It is making me blue that the points in the graph aren't blue :`(
+
+ANSWER: The points were not blue because the code was trying to manipulate the color as an aesthetic -- that is to say, within aes(), but manipulating the color of the dots based on something other than a variable is an **attibute** and thus should be outside aes(). I have made the proper change below:
+
+```{r}
+ggplot(data = mpg) + 
+  geom_point(mapping = aes(x = displ, y = hwy), color = "blue")
+
+```
+
+
+
+### Graph Fail 3
+Two mistakes in this graph. First, I wanted to make the the points slightly bolder, but changing the alpha to 2 does nothing. What does alpha do and what does setting it to 2 do? What could be done instead if I want the points slightly bigger? 
+
+ANSWER: alpha manipulates transparency, not size. And the reason it was not altered is that alpha should be between 0 and 1. The code probably reads any number above 1 as equivalent to 1. If you want the points to be bigger, you should manipulate size, which I did below.
+
+Second, I wanted to move the legend on top of the graph since there aren't any points there, putting it at approximately the point/ordered pair (5, 40). How do you actually do this? Also, how do you remove the legend title ("class")? Finally, how would you remove the plot legend completely?
+
+ANSWER: To move the legend to the top, set legend.position equal to "top", which I did below. To remove the legend title "class", I added a new theme line that sets legend.title to element_blank(). If I wanted to remove the legend completely, I could set legend.position equal to "none".
+
+```{r}
+mpg %>% 
+ggplot() + 
+  geom_point(mapping = aes(x = displ, y = hwy, color = class), alpha = 1, size = 3) + 
+  theme(legend.direction = "horizontal") + 
+  theme(legend.position = "top") +
```

Comment:
While you did move the legend to the top of the figure, I interpreted the request as asking for the legend to be moved to the top of within the graph (@ around the coordinates of (5, 40), so the way I did this was by setting the position of the legend to be based off of relative positions within the graph with legend.position:
`theme(legend.direction = "horizontal", 
        legend.position = c(.6, .85), 
        legend.title =   element_blank())`